lang/src/org/partiql/lang/compiler/PartiQLCompilerDefault.kt

```diff
@@ -56,14 +61,22 @@ internal class PartiQLCompilerDefault(
         relationalOperatorFactory = operatorFactories
     )
 
+    // Initialize a map from a type alias to a scalar type, as part of work for PartiQL compiler to install the plugin
+    private val aliasToScalarType: Map<String, ScalarType>
+
     init {
+        plugin.validate()
+
+        aliasToScalarType = plugin.mapAliasToScalarType()
+
         exprConverter = PhysicalPlanCompilerImpl(
             valueFactory = valueFactory,
             functions = functions,
             customTypedOpParameters = customTypedOpParameters,
             procedures = procedures,
             evaluatorOptions = evaluatorOptions,
-            bexperConverter = bexprConverter
+            bexperConverter = bexprConverter,
+            aliasToScalarType = aliasToScalarType
```

Comment:
Same as before. We don't have to pass around List<T> and Map<K, V> everywhere. It's generally better to wrap these things to encapsulate the underlying data structure — which we are then free to change.

Think of the type identifiers to type names mappings. If we have `Map<String, String>` then we are limited to just looking up the id of given a name. We can do this and more by wrapping. Call it the TypeRegistry or something (don't use this name, this is just an example for illustrative purposes).

```
class TypeRegistry(private val types: List<StaticType>) {

  private lateinit var namesToType: Map<String, StaticType>
  private lateinit var idToType: Map<String, StaticType>

  init {
    // populate the maps
  }

  // now we control access to the underlying data structures
  fun getTypeByName(name: String): StaticType = namesToType[name]!! // something better in practice
 
  // contrived example
  fun getTypeById(id: String): StaticType = idsToType[id]!! 

 // can be extended and different underlying data structures changed
}
```